### ISSUE TITLE
only limited number of users are shown from LDAP address book

### ISSUE THREAD

**Issue by atroxix** [owncloud/contacts#1094]
I've connected LDAP address book to Active Directory and there are at least several hundred users, but only about 200 (even less) are listed when browsing.

Using OwnCloud 8.2 and Contacts 0.5.0.0 (was the same with OwnCloud 8.1 and Contacts 0.4.0.x).

**Comment by babelouest** [comment]
There is an arbitrary limit in the ldap contact search of 10 pages.
As a workaround, you can change the page size value in the addressbook settings.

**Comment by atroxix** [comment]
The default setting for the page size is 20, but in my case all contacts are shown on one page and I'm not able to change LDAP address book settings currently (because of another bug which I can point to later).

**Comment by babelouest** [comment]
Yes, the page size is a LDAP search parameter, and the page number is set to 10 in the source code, so in your case, it makes sense to have no more than 200 contacts.

I remember, when I was trying to select a huge ldap addressbook without limits, there was a server timeout and/or a memory limit reached.

**Comment by atroxix** [comment]
Maybe its possible to select only a number of LDAP objects, something like LIMIT keyword works in SQL?

**Comment by babelouest** [comment]
that's the point of the page size parameter (not very understandable at first I agree)

**Comment by atroxix** [comment]
Then it doesn't work properly I guess? Because in my case there's only 1 page with a bit less than 200 contacts (or maybe it's even 200 exactly - I haven't counted fully) and no more pages or anything.

**Comment by babelouest** [comment]
the `page size` parameter isn't the number of contacts shown in Owncloud UI, it's the number of ldap contacts returned in a ldap search query.

Since ldap backend can pipe no more than 10 search result lists, in your case, your UI will display 20 (page size)*10 (search query limit) = 200 contacts.

**Comment by atroxix** [comment]
Can this option be renamed? :-)

I most set page size to <number of my contacts> / 10 + some extra?

**Comment by atroxix** [comment]
I raised this hard-coded limit of 10 to 100 now, and I see now 2000 contacts.

Can some pagination be implemented, so that this "page size" would really mean how many contacts are queried and shown per page in address book?

**Comment by babelouest** [comment]
There is an issue to implement the pagination but I can't find it.
I'll add to the config popup the number of search in the loop and rename the page size parameter. But the pagination can't be implemented in the ldap backend, only in contacts UI source code.

**Comment by atroxix** [comment]
I guess this is it: #136 (unfortunately closed).

Although there would be little use of pagination if it's for UI only.
